loopdb/store.go

```diff
@@ -227,59 +244,68 @@ func deserializeUpdates(swapBucket *bbolt.Bucket) ([]*LoopEvent, error) {
 	return updates, nil
 }
 
-// FetchLoopOutSwaps returns all loop out swaps currently in the store.
+// FetchLoopInSwaps returns all loop in swaps currently in the store.
 //
 // NOTE: Part of the loopdb.SwapStore interface.
-func (s *boltSwapStore) FetchLoopOutSwaps() ([]*LoopOut, error) {
-	var swaps []*LoopOut
-
-	err := s.fetchSwaps(loopOutBucketKey,
-		func(contractBytes []byte, loop Loop) error {
-			contract, err := deserializeLoopOutContract(
-				contractBytes, s.chainParams,
-			)
-			if err != nil {
-				return err
-			}
+func (s *boltSwapStore) FetchLoopInSwaps() ([]*LoopIn, error) {
+	var swaps []*LoopIn
 
-			swaps = append(swaps, &LoopOut{
-				Contract: contract,
-				Loop:     loop,
-			})
+	err := s.db.View(func(tx *bbolt.Tx) error {
+		// First, we'll grab our main loop in bucket key.
+		rootBucket := tx.Bucket(loopInBucketKey)
+		if rootBucket == nil {
+			return errors.New("bucket does not exist")
+		}
 
-			return nil
-		},
-	)
-	if err != nil {
-		return nil, err
-	}
+		// We'll now traverse the root bucket for all active swaps. The
+		// primary key is the swap hash itself.
+		return rootBucket.ForEach(func(swapHash, v []byte) error {
+			// Only go into things that we know are sub-bucket
+			// keys.
+			if v != nil {
```

Comment:
Not part of this review, just leaving a side note for the future: would be nice to have a sub bucket containing the swap buckets instead.

Reply:
How would that be advantageous? Maybe if we need to add loop-type-specific data that isn't related to a particular swap? Workaround is to just create a new top level bucket. I wouldn't really want to store anything other than swap buckets in the `rootBucket` in this function.

Reply:
Yes, true. I missed the part this is not the root bucket but a "root bucket". In this case it really is not of any advantage.

Reply:
Hehe, yes it is the local root :)